qbft/decided.go

```diff
@@ -34,10 +34,10 @@ func (c *Controller) UponDecided(msg *SignedMessage) (*SignedMessage, error) {
 	// Mark current instance decided
 	if inst := c.InstanceForHeight(c.Height); inst != nil && !inst.State.Decided {
 		inst.State.Decided = true
-		if msg.Message.Round > inst.State.Round {
-			inst.State.Round = msg.Message.Round
-		}
 		if c.Height == msg.Message.Height {
+			if msg.Message.Round > inst.State.Round {
```

Comment:
we can remove if, @amir-blox please add spec tests for cases that round < state round, round == state round and round > state round

Reply:
The tests already exists:
`decided.CurrentInstance()`
`decided.CurrentInstanceFutureRound()`
`decided.CurrentInstancePastRound()` - fixed ctrl root as now we use the right round